Python/symtable.c

```diff
@@ -1488,7 +1488,8 @@ symtable_extend_namedexpr_scope(struct symtable *st, expr_ty e)
          */
         if (ste->ste_comprehension) {
             long target_in_scope = _PyST_GetSymbol(ste, target_name);
-            if (target_in_scope & DEF_COMP_ITER) {
+            if ((target_in_scope & DEF_COMP_ITER) &&
+                (target_in_scope & DEF_LOCAL)) {
```

Comment:
```suggestion
            if (target_in_scope & (DEF_COMP_ITER | DEF_LOCAL)) {

```

Reply:
Tests do not pass with this change 😥

Reply:
Whoops, I missed the ` == (DEF_COMP_ITER | DEF_LOCAL)` at the end of the suggestion 😅

Reply:
```suggestion
            if (target_in_scope & (DEF_COMP_ITER | DEF_LOCAL) == (DEF_COMP_ITER | DEF_LOCAL)) {

```

If you find it less clear feel free to skip this suggestion

Reply:
Yeah, I would prefer to stick to my version :)
Thanks for the suggestion, though!